transmart-rest-api/grails-app/services/org/transmartproject/rest/dataExport/DataExportService.groovy

```diff
@@ -0,0 +1,80 @@
+package org.transmartproject.rest.dataExport
+
+import grails.transaction.Transactional
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import org.transmartproject.core.exceptions.InvalidRequestException
+import org.transmartproject.core.exceptions.LegacyStudyException
+import org.transmartproject.core.users.ProtectedOperation
+import org.transmartproject.core.users.User
+import org.transmartproject.db.multidimquery.query.Constraint
+import org.transmartproject.db.multidimquery.query.OrConstraint
+import org.transmartproject.db.multidimquery.query.PatientSetConstraint
+import org.transmartproject.rest.MultidimensionalDataService
+import org.transmartproject.rest.MultidimensionalDataService.Format as Format
+import sun.reflect.generics.reflectiveObjects.NotImplementedException
+
+import java.util.zip.ZipOutputStream
+
+@Transactional
+@Component("restDataExportService")
+class DataExportService {
+
+    @Autowired
+    MultidimensionalDataService multidimensionalDataService
+
+    private static Constraint patientSetConstraint(List<Long> ids) {
+        new OrConstraint(args:  ids.collect { new PatientSetConstraint(patientSetId: it) } )
+    }
+
+    List highDimDataTypesForPatientSets(List<Long> patientSetIds, User user) {
+        def constraint = patientSetConstraint(patientSetIds)
+        highDimDataTypes(constraint, user)
+    }
+
+    List highDimDataTypesForObservationSets(List patientSetIds, User user) {
+        //TODO saving observationSets is not supported yet
+        throw new NotImplementedException()
+    }
+
+    private List highDimDataTypes(Constraint constraint, User user) {
+        multidimensionalDataService.multiDimService.retriveHighDimDataTypes(constraint, user)
+    }
+
+    List patientSetsExportPermission(List<Long> ids, User user){
+        ids.collect { setId ->
+            multidimensionalDataService.multiDimService.findPatientSet(setId, user, ProtectedOperation.WellKnownOperations.EXPORT)
```

Comment:
So why EXPORT is one of ProtectedOperation.WellKnownOperations? Isn't it different access level?